File: CourierAmexClient/src/app/features/company/models/supplier.model.ts
import { BaseEntity } from "@app/models";
import { LocationModel } from "./location.model";

export interface SupplierModel extends BaseEntity<number> {
  name: string;
  companyId?: number;
  countryId?: number;
  address?: string;
  phone?: string;
  contact?: string;
  locations: LocationModel[];

  companyName?: string;
  countryName?: string;
}

export const newSupplierModel = {
  id: 0,
  companyId: undefined,
  countryId: undefined,
  name: '',
  status: 2,
  locations: []
};

File: CourierAmexAPI/CourierAmex/Models/SupplierModel.cs
namespace CourierAmex.Models
{
    public class SupplierModel : BaseEntityModel<int>
    {
        public int CompanyId { get; set; }
        public string Name { get; set; }
        public short CountryId { get; set; }
        public string? Address{ get; set; }
        public string? Phone { get; set; }
        public string? Contact { get; set; }
        public List<LocationModel> Locations { get; set; }

        public string? CompanyName { get; set; }
        public string? CountryName { get; set; }
    }
}
